huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_plugin.go

```diff
@@ -0,0 +1,320 @@
+	deleteRabbitmqPluginPath = strings.ReplaceAll(deleteRabbitmqPluginPath, "{project_id}", deleteRabbitmqPluginClient.ProjectID)
+	deleteRabbitmqPluginPath = strings.ReplaceAll(deleteRabbitmqPluginPath, "{instance_id}", instanceID)
+	jsonBody := utils.RemoveNil(buildRabbitmqPluginBody(d, false))
+
+	reqOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         jsonBody,
+	}
+
+	retryFunc := func() (interface{}, bool, error) {
+		_, err = deleteRabbitmqPluginClient.Request("PUT", deleteRabbitmqPluginPath, &reqOpt)
+		retry, err := handleMultiOperationsError(err)
+		return nil, retry, err
+	}
+	_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     rabbitmqInstanceStateRefreshFunc(deleteRabbitmqPluginClient, instanceID),
+		WaitTarget:   []string{"RUNNING"},
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		DelayTimeout: 1 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return diag.Errorf("error disabling the plugin: %s", err)
+	}
+
+	// The RabbitMQ disabling plugin is done if the status of its task is SUCCESS.
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"CREATED"},
+		Target:       []string{"SUCCESS"},
+		Refresh:      rabbitmqPluginTaskRefreshFunc(deleteRabbitmqPluginClient, instanceID, d, "rabbitmqPluginModify"),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        1 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the RabbitMQ (%s) disabling plugin (%s) to be done: %s", instanceID, d.Get("name"), err)
+	}
+
+	return resourceDmsRabbitmqPluginRead(ctx, d, cfg)
+}
+
+func rabbitmqPluginTaskRefreshFunc(client *golangsdk.ServiceClient, instanceID string, d *schema.ResourceData,
+	taskName string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		// getRabbitmqPluginTask: get RabbitMQ plugin task
+		getRabbitmqPluginTaskHttpUrl := "v2/{project_id}/instances/{instance_id}/tasks"
+		getRabbitmqPluginTaskPath := client.Endpoint + getRabbitmqPluginTaskHttpUrl
+		getRabbitmqPluginTaskPath = strings.ReplaceAll(getRabbitmqPluginTaskPath, "{project_id}",
+			client.ProjectID)
+		getRabbitmqPluginTaskPath = strings.ReplaceAll(getRabbitmqPluginTaskPath, "{instance_id}", instanceID)
+
+		reqOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+		}
+		getRabbitmqPluginTaskResp, err := client.Request("GET", getRabbitmqPluginTaskPath, &reqOpt)
+
+		if err != nil {
+			return nil, "QUERY ERROR", err
+		}
+
+		getRabbitmqPluginTaskRespBody, err := utils.FlattenResponse(getRabbitmqPluginTaskResp)
+		if err != nil {
+			return nil, "PARSE ERROR", err
+		}
+
+		task := getRabbitmqPluginTask(taskName, d, getRabbitmqPluginTaskRespBody)
+		if task == nil {
+			return nil, "NIL ERROR", fmt.Errorf("can not find the task")
```

Comment:
`NIL ERROR` is hard to understand and when the error is not nil, it will arise the error message regardless of the status.

so I think we can just set the status to **ERROR**.
```suggestion
			return nil, "ERROR", fmt.Errorf("can not find the task")
```

Reply:
done